hoomd/md/EvaluatorPairLJ0804.h

```diff
@@ -0,0 +1,233 @@
+    equivalent to inlining code, so there is no performance loss.
+
+    In hoomd, a "standard" pair potential is defined as V(rsq, rcutsq, params, di, dj, qi, qj), where rsq is the squared
+    distance between the two particles, rcutsq is the cutoff radius at which the potential goes to 0, params is any
+    number of per type-pair parameters, di, dj are the diameters of particles i and j, and qi, qj are the charges of
+    particles i and j respectively.
+
+    Diameter and charge are not always needed by a given pair evaluator, so it must provide the functions
+    needsDiameter() and needsCharge() which return boolean values signifying if they need those quantities or not. A
+    false return value notifies PotentialPair that it need not even load those values from memory, boosting performance.
+
+    If needsDiameter() returns true, a setDiameter(Scalar di, Scalar dj) method will be called to set the two diameters.
+    Similarly, if needsCharge() returns true, a setCharge(Scalar qi, Scalar qj) method will be called to set the two
+    charges.
+
+    All other arguments are common among all pair potentials and passed into the constructor. Coefficients are handled
+    in a special way: the pair evaluator class (and PotentialPair) manage only a single parameter variable for each
+    type pair. Pair potentials that need more than 1 parameter can specify that their param_type be a compound
+    structure and reference that. For coalesced read performance on G200 GPUs, it is highly recommended that param_type
+    is one of the following types: Scalar, Scalar2, Scalar4.
+
+    The program flow will proceed like this: When a potential between a pair of particles is to be evaluated, a
+    PairEvaluator is instantiated, passing the common parameters to the constructor and calling setDiameter() and/or
+    setCharge() if need be. Then, the evalForceAndEnergy() method is called to evaluate the force and energy (more
+    on that later). Thus, the evaluator must save all of the values it needs to compute the force and energy in member
+    variables.
+
+    evalForceAndEnergy() makes the necessary computations and sets the out parameters with the computed values.
+    Specifically after the method completes, \a force_divr must be set to the value
+    \f$ -\frac{1}{r}\frac{\partial V}{\partial r}\f$ and \a pair_eng must be set to the value \f$ V(r) \f$ if \a energy_shift is false or
+    \f$ V(r) - V(r_{\mathrm{cut}}) \f$ if \a energy_shift is true.
+
+    A pair potential evaluator class is also used on the GPU. So all of its members must be declared with the
+    DEVICE keyword before them to mark them __device__ when compiling in nvcc and blank otherwise. If any other code
+    needs to diverge between the host and device (i.e., to use a special math function like __powf on the device), it
+    can similarly be put inside an ifdef __HIPCC__ block.
+
+    <b>LJ specifics</b>
+
+    EvaluatorPairLJ0804 evaluates the function:
```

Comment:
```suggestion
/*! EvaluatorPairLJ0804 evaluates the function:
```
There is no need to duplicate the documentation from `EvaluatorPairLJ.h`